extensions/windows-event-log/wel/MetadataWalker.h

```diff
@@ -95,16 +90,12 @@ class MetadataWalker : public pugi::xml_tree_walker {
 
   const WindowsEventLogMetadata& windows_event_log_metadata_;
   std::string log_name_;
-  std::optional<utils::Regex> regex_;
+  const std::optional<utils::Regex>& regex_;
```

Comment:
I feel like a nullable raw pointer to regex is similar enough that it probably fulfills the requirements, but it's a simpler representation, so I would rather go with that if it works. But only if this class doesn't need to react to concurrent changes of the regex data member.

Reply:
Makes sense 👍 and fortunetly the usage of the Metadawalker is contained so using raw ptr looks ok, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1434/commits/e142a7d9796ac63ccd4758a9e5a570dae1509fc9#diff-ce0af7f15ffcce3655d42459495774ad88e0f15a938824ec5ca733d30d18c704R93